site/content/contribute/server/style-guide.md

```diff
@@ -0,0 +1,130 @@
+
+Another common pattern is to preemptively declare the interface in the source package itself, so that the consumer can just directly import the interface. Instead, try to declare the interface in the package which is going to consume the functionality. Often, different packages have non-overlapping set of functionalities to consume. If you do find several consumers of the package, remember that interfaces can be composed. So define small chunks of functionalities in different interfaces, and let consumers compose them as needed. Take a look at the set of interfaces in the [io](https://golang.org/pkg/io/) package.
+
+These are just guidelines and not strict rules. Understand your use-case and apply them appropriately.
+
+## Stylistic
+
+### [CamelCase variables/constants](https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps)
+
+We use CamelCase names like WebsocketEventPostEdited, not WEBSOCKET_EVENT_POST_EDITED.
+
+This rule is not yet fully applied to the `model` package due to backwards compatibility requirements.
+
+### Empty string check
+
+Use `foo == ""` to check for empty strings, not `len(foo) == 0`.
+
+### [Reduce indentation](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow)
+
+If there are multiple return statements in an if-else statement, remove the else block and outdent it.
+
+This is an example from `mlog/human/parser.go`:
+
+```go
+// Look for an initial "{"
+if token, err := dec.Token(); err != nil {
+	return result, err
+} else {
+	d, ok := token.(json.Delim)
+	if !ok || d != '{' {
+		return result, fmt.Errorf("input is not a JSON object, found: %v", token)
+	}
+}
+```
+
+This can be simplified to:
+
+```go
+// Look for an initial "{"
+if token, err := dec.Token(); err != nil {
+	return result, err
+}
+d, ok := token.(json.Delim)
+if !ok || d != '{' {
+	return result, fmt.Errorf("input is not a JSON object, found: %v", token)
+}
+```
+
+### [Initialisms](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)
+
+Use `userID` rather than `userId`. Same for abbreviations; `HTTP` is preferred over `Http` or `http`.
+
+This rule is not yet fully applied to the `model` package due to backwards compatibility requirements.
+
+### [Receiver Names](https://github.com/golang/go/wiki/CodeReviewComments#receiver-names)
+
+The name of a method's receiver should be a reflection of its identity; often a one or two letter abbreviation of its type suffices (such as "c" or "cl" for "Client"). Don't use generic names such as "me", "this" or "self", identifiers typical of object-oriented languages that give the method a special meaning.
```

Comment:
Slightly confused about what "give the method a special meaning" here means. Does it mean that they distinguish between a "method" and a "function", or that they give "self" or "this" a special meaning?

Reply:
The latter. Happy to rephrase in some other way that you prefer.

Reply:
```suggestion
The name of a method's receiver should be a reflection of its identity; often a one or two letter abbreviation of its type suffices (such as "c" or "cl" for "Client"). Don't use generic names such as "me", "this" or "self", identifiers typical of object-oriented languages that give the variable a special meaning.
```

Best I can come up with. What do you think?

Reply:
Works for me!